### PR TITLE
Scpui ship select

### PR DESCRIPTION
**Pull request by MjnMixael**
Gives Lua access to the loadout wing slots. I had considerable trouble with this one due to lack of coding skill. Open to suggestions.

This is the first PR of several required to get ship and weapon select fully functional.

### CODE REVIEW

File: code/scripting/api/libs/ui.cpp [hunk 1]
```diff
@@ -1036,6 +1038,71 @@ ADE_FUNC(getFictionMusicName, l_UserInterface_FictionViewer, nullptr,
...
+ADE_LIB_DERIV(l_Loadout_Wings, "Loadout_Wings", nullptr, nullptr, l_UserInterface_ShipWepSelect);
+ADE_INDEXER(l_Loadout_Wings, //
+	"number Index",
+	"Array of loadout wing data",
+	"loadout_wing",
+	"loadout handle, or invalid handle if index is invalid")
+{
+	int idx;
+	if (!ade_get_args(L, "*i", &idx))
+		return ade_set_error(L, "s", "");
```
Comment (BMagnu) [inline code/scripting/api/libs/ui.cpp:1096]: Unlike what the docs say, you're not returning an invalid handle (which'd be ``ade_set_error(L,"o", l_Loadout_Wing.Set(ss_wing_info_h()));``), but an empty string

**Review by BMagnu — CHANGES_REQUESTED** [review]
One small comment

File: code/scripting/api/libs/ui.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/scripting/api/libs/ui.cpp]: Fixed

Comment (BMagnu) [reply code/scripting/api/libs/ui.cpp]: I'm reasonably sure that you still have to _return_ the ``ade_set_error`` there, not just call it.
But yeah, that'll be doing a proper invalid handle

Comment (MjnMixael) [reply code/scripting/api/libs/ui.cpp]: Whoops. Didn't intend to replace the return. Fixed

**Review by BMagnu — APPROVED** [review]

**Pull request merged**